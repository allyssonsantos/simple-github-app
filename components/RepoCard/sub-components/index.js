import Header from './Header';
import Content from './Content';
import Footer from './Footer';

export { Header, Content, Footer };
