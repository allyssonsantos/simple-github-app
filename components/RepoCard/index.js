import React, { Component } from 'react';
import styled from 'styled-components';

import { Header, Content, Footer } from './sub-components';

const StyledCard = styled.article`
  background-color: #fff;
  border-radius: 3px;
  box-shadow: 0px 0px 1px #000;
  cursor: pointer;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  min-height: 150px;
  padding: 10px;
`;

class RepoCard extends Component {
  constructor(props) {
    super(props);
  }

  static Header = Header;
  static Content = Content;
  static Footer = Footer;

  render() {
    const { children, onClick } = this.props;
    return <StyledCard onClick={onClick}>{children}</StyledCard>;
  }
}

RepoCard.displayName = 'RepoCard';
RepoCard.Header.displayName = 'RepoCard.Header';
RepoCard.Content.displayName = 'RepoCard.Content';
RepoCard.Footer.displayName = 'RepoCard.Footer';

export default RepoCard;
