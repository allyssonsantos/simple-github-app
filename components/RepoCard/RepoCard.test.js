import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import RepoCard from './';

describe('<RepoCard />', () => {
  describe('With Snapshot Testing', () => {
    it('Should match snapshot without compound components children', () => {
      const component = renderer.create(
        <RepoCard>This is a RepoCard</RepoCard>
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('Should match snapshot Header, Content and Footer children components', () => {
      const PanelComponent = () => (
        <RepoCard>
          <RepoCard.Header>Header</RepoCard.Header>
          <RepoCard.Content>Content</RepoCard.Content>
          <RepoCard.Footer stars="10" updated="">
            Footer
          </RepoCard.Footer>
        </RepoCard>
      );
      const component = renderer.create(<PanelComponent />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Date function', () => {
    it('Should <RepoCard.Footer> format correct date', () => {
      const component = shallow(
        <RepoCard.Footer updated="2019-01-08T10:50:21Z" />
      );
      const expectedFormatedDate = 'Last update: 01/08/2019';
      expect(component.find('Updated').text()).toBe(expectedFormatedDate);
    });
  });
});
