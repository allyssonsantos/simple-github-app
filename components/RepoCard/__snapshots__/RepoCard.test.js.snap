// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoCard /> With Snapshot Testing Should match snapshot Header, Content and Footer children components 1`] = `
.c1 {
  color: #000;
  font-size: 16px;
  margin: 0;
  margin-bottom: 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 {
  background-color: #fff;
  border-radius: 3px;
  box-shadow: 0px 0px 1px #000;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-height: 150px;
  padding: 10px;
}

<article
  className="c0"
>
  <header>
    <h2
      className="c1"
      size="small"
    >
      Header
    </h2>
  </header>
  <div>
    Content
  </div>
  <footer
    className="c2"
  >
    <span
      className=""
    >
      Stars: 
      10
    </span>
    <span
      className=""
    >
      Last update: 
      
      /
      
      /
      
    </span>
  </footer>
</article>
`;

exports[`<RepoCard /> With Snapshot Testing Should match snapshot without compound components children 1`] = `
.c0 {
  background-color: #fff;
  border-radius: 3px;
  box-shadow: 0px 0px 1px #000;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-height: 150px;
  padding: 10px;
}

<article
  className="c0"
>
  This is a RepoCard
</article>
`;
